GMX-Bots/Large-Open-and-Increase-Positions/src/agent.spec.ts

```diff
@@ -0,0 +1,549 @@
+    // multiple findings for increasePosition event
+    expect(findings).toStrictEqual([
+      mockCreateFinding(
+        MOCK_ACCOUNT,
+        mockNetworkManager.vaultAddress,
+        key,
+        key,
+        size1,
+        sizeDelta1,
+        ethers.BigNumber.from(size1).sub(ethers.BigNumber.from(sizeDelta1))
+      ),
+      mockCreateFinding(
+        MOCK_ACCOUNT,
+        mockNetworkManager.vaultAddress,
+        key,
+        key,
+        size2,
+        sizeDelta2,
+        ethers.BigNumber.from(size2).sub(ethers.BigNumber.from(sizeDelta2))
+      ),
+      mockCreateFinding(
+        MOCK_ACCOUNT,
+        mockNetworkManager.vaultAddress,
+        key,
+        key,
+        size3,
+        sizeDelta3,
+        ethers.BigNumber.from(size3).sub(ethers.BigNumber.from(sizeDelta3))
+      ),
+    ]);
+    expect(findings.length).toStrictEqual(3);
+
+    const increasePositionEventLog4 = MOCK_IFACE.encodeEventLog(increasePositionEventFragment, [
+      key,
+      MOCK_ACCOUNT,
+      MOCK_TOKEN,
+      MOCK_TOKEN,
+      0,
+      sizeDelta4,
+      true,
+      0,
+      0,
+    ]);
+
+    const updatePositionEventLog4 = MOCK_IFACE.encodeEventLog(updatePositionEventFragment, [key, size4, 0, 0, 0, 0, 0]);
+
+    const increasePositionEventLog5 = MOCK_IFACE.encodeEventLog(increasePositionEventFragment, [
+      key,
+      MOCK_ACCOUNT,
+      MOCK_TOKEN,
+      MOCK_TOKEN,
+      0,
+      sizeDelta5,
+      true,
+      0,
+      0,
+    ]);
+
+    const updatePositionEventLog5 = MOCK_IFACE.encodeEventLog(updatePositionEventFragment, [key, size6, 0, 0, 0, 0, 0]);
+
+    const increasePositionEventLog6 = MOCK_IFACE.encodeEventLog(increasePositionEventFragment, [
+      key,
+      MOCK_ACCOUNT,
+      MOCK_TOKEN,
+      MOCK_TOKEN,
+      0,
+      sizeDelta6,
+      true,
+      0,
+      0,
+    ]);
+
+    const updatePositionEventLog6 = MOCK_IFACE.encodeEventLog(updatePositionEventFragment, [key, size5, 0, 0, 0, 0, 0]);
+
+    txEvent = new TestTransactionEvent()
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        increasePositionEventLog4.data,
+        ...increasePositionEventLog4.topics
+      )
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        updatePositionEventLog4.data,
+        ...updatePositionEventLog4.topics
+      )
+
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        increasePositionEventLog5.data,
+        ...increasePositionEventLog5.topics
+      )
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        updatePositionEventLog5.data,
+        ...updatePositionEventLog5.topics
+      )
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        increasePositionEventLog6.data,
+        ...increasePositionEventLog5.topics
+      )
+      .addAnonymousEventLog(
+        mockNetworkManager.vaultAddress,
+        updatePositionEventLog6.data,
+        ...updatePositionEventLog5.topics
```

Comment:
It's disturbing why you had to use numbers in non-organized way. 

You're giving `size6` to `updatePositionEventLog5` and `size5` to `updatePositionEventLog6`, then you switched them here again. Please fix that for a better code readability.